synapse/lib/aha.py

```diff
@@ -555,6 +644,128 @@ async def addAhaSvc(self, name, info, network=None):
         # mostly for testing...
         await self.fire('aha:svcadd', svcinfo=svcinfo)
 
+    def _getAhaName(self, name):
+        # the modern version of names is absolute or ...
+        if name.endswith('...'):
+            netw = self.conf.get('aha:network')
+            name = name[:-2] + netw
+        return name
+
+    async def getAhaPool(self, name):
+        return self._getAhaPool(name)
+
+    def _getAhaPool(self, name):
+        name = self._getAhaName(name)
+        byts = self.slab.get(name.encode(), db='aha:pools')
+        if byts is not None:
+            return s_msgpack.un(byts)
+
+    def _savePoolInfo(self, poolinfo):
+        reqValidPoolInfo(poolinfo)
+        name = poolinfo.get('name')
+        self.slab.put(name.encode(), s_msgpack.en(poolinfo), db='aha:pools')
+
+    def _loadPoolInfo(self, name):
+        byts = self.slab.get(name.encode(), db='aha:pools')
+        if byts is not None:
+            return s_msgpack.un(byts)
+
+    def _reqPoolInfo(self, name):
+
+        poolinfo = self._loadPoolInfo(name)
+        if poolinfo is not None:
+            return poolinfo
+
+        mesg = f'There is no AHA service pool named {name}.'
+        raise s_exc.NoSuchName(mesg=mesg)
+
+    async def addAhaPool(self, name, info):
+
+        name = self._getAhaName(name)
+
+        if await self._getAhaSvc(name) is not None:
+            mesg = f'An AHA service or pool is already using the name "{name}".'
+            raise s_exc.DupName(mesg=mesg)
+
+        info['name'] = name
+        info['created'] = s_common.now()
+        info['services'] = {}
+
+        info.setdefault('creator', self.getDmonUser())
+
+        return await self._push('aha:pool:add', info)
+
+    async def getAhaPools(self):
+        for lkey, byts in self.slab.scanByFull(db='aha:pools'):
+            yield s_msgpack.un(byts)
+
+    @s_nexus.Pusher.onPush('aha:pool:add')
+    async def _addAhaPool(self, info):
+        self._savePoolInfo(info)
+        return info
+
+    async def addAhaPoolSvc(self, poolname, svcname, info):
+        info['created'] = s_common.now()
+        info.setdefault('creator', self.getDmonUser())
+        return await self._addAhaPoolSvc(poolname, svcname, info)
```

Comment:
Could this just push a `aha:pool:svc:add` and then change `_addAhaPoolSvc` to `onPush` instead of `onPushAuto`?